lib/src/request.ts

```diff
@@ -0,0 +1,10 @@
+/**
+ * Common Media Request
+ * 
+ * @packageDocumentation
+ */
+export type { ResourceTiming } from './request/ResourceTiming.js';
+export type { CommonMediaRequest } from './request/CommonMediaRequest.js';
+export type { CommonMediaResponse } from './request/CommonMediaResponse.js';
+export type { RequestPlugin } from './request/RequestPlugin.js';
+export type { ResponsePlugin } from './request/ResponsePlugin.js';
```

Comment:
Is "plugin" the right terminology here? Would `RequestModifier` or `RequestTransform` be more appropriate?

Reply:
I agree, "plugin" is not the most suitable terminology.
What about `RequestInterceptor` and `ResponseInterceptor`? (as in Android OkHttp client)

Reply:
I'm not familiar with that client, but interceptor is the name used on chromecast to describe similar functionality:

https://developers.google.com/cast/docs/reference/web_receiver/cast.framework.PlayerManager#setMessageInterceptor

This brings up the question: How should the `null` use case be handled? (an interceptor returning a nullish value)

Reply:
I've renamed.
`null` use case could be used for enabling the application to force not sending the request, but I don't see the point of doing so.
Should we consider this possiblity ?
As it is (`(request: CommonMediaRequest) => Promise<CommonMediaRequest>`), the interceptor has to return a Promise resolved with a `CommongMediaRequest` object